Volto 15.4.x no longer supports Language Variants on the backend (en_US, nl_BE, etc.)

When you create a new API backend for a Volto site in Plone 6 Alpha 4 the  Language dropdown contains both Languages (English, Dutch) and Language Variants, like English (USA), Dutch (Belgium).  The former have codes like EN and NL, the latter are encoded with en_US and nl_BE . 

The Default selected language depends on your locale and which browser you use. Some uses with their local and browser will have a language variant selected by default. This explains why not everybody might see this issue. 

<img width="593" alt="Screenshot 2022-05-02 at 16 06 20" src="https://user-images.githubusercontent.com/394784/166247639-dbe5da99-0667-4977-8e68-667b20ee05f1.png">

Volto 15.4.1 will throw an error if you created the backedn with any Language Variant:

```
"Unhandled Rejection (Error): Cannot find module './en_US.json'
./locales lazy recursive ^./..json$/webpackAsyncContext/<
/home/Code/Plone/Plone-6/frontend/locales lazy ^/./.json$ namespace object:50"
```

This appears to be a regression, as the documentation explicitly has a language variant as an example and Plone 5.2 also has language variants enabled and listed in the Site language dropdown when you create a site.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -15,6 +15,7 @@
 - Fix toolbar bottom opacity on edit page when toolbar is collapsed @ichim-david
 - Fixed secure cookie option. @giuliaghisini
 - Changed addon order in addon controlpanel to mimic Classic UI @erral
+- Fixed error when loading content in a language for which a Volto translation is not available. @davisagli
 
 ### Internal
 

--- a/src/middleware/api.js
+++ b/src/middleware/api.js
@@ -188,7 +188,12 @@ export default (api) => ({ dispatch, getState }) => (next) => (action) => {
         }
         if (type === GET_CONTENT) {
           const lang = result?.language?.token;
-          if (lang && getState().intl.language !== lang && !subrequest) {
+          if (
+            lang &&
+            getState().intl.language !== lang &&
+            !subrequest &&
+            config.settings.supportedLanguages.includes(lang)
+          ) {
             const langFileName = normalizeLanguageName(lang);
             import('~/../locales/' + langFileName + '.json').then((locale) => {
               dispatch(changeLanguage(lang, locale.default));

--- a/src/server.jsx
+++ b/src/server.jsx
@@ -236,7 +236,10 @@ server.get('/*', (req, res) => {
         req.headers['accept-language'];
 
       if (cookie_lang !== contentLang) {
-        store.dispatch(changeLanguage(contentLang, locales[contentLang], req));
+        const newLocale = new locale.Locales(contentLang)
+          .best(supported)
+          .toString();
+        store.dispatch(changeLanguage(newLocale, locales[newLocale], req));
       }
 
       const context = {};

